pkg/network/usm/procnet/parser.go

```diff
@@ -0,0 +1,223 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package procnet
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"net/netip"
+	"os"
+	"slices"
+	"strconv"
+)
+
+// This file contains the code that does the parsing of /proc/net/{tcp,tcp6} files.
+//
+// The format of these files is the following:
+//
+//    46: 010310AC:9C4C 030310AC:1770 01
+//    |      |      |      |      |   |--> connection state
+//    |      |      |      |      |------> remote TCP port number
+//    |      |      |      |-------------> remote IPv4 address
+//    |      |      |--------------------> local TCP port number
+//    |      |---------------------------> local IPv4 address
+//    |----------------------------------> number of entry
+//
+//    00000150:00000000 01:00000019 00000000
+//       |        |     |     |       |--> number of unrecovered RTO timeouts
+//       |        |     |     |----------> number of jiffies until timer expires
+//       |        |     |----------------> timer_active (see below)
+//       |        |----------------------> receive-queue
+//       |-------------------------------> transmit-queue
+//
+//    1000        0 54165785 4 cd1e6040 25 4 27 3 -1
+//     |          |    |     |    |     |  | |  | |--> slow start size threshold,
+//     |          |    |     |    |     |  | |  |      or -1 if the threshold
+//     |          |    |     |    |     |  | |  |      is >= 0xFFFF
+//     |          |    |     |    |     |  | |  |----> sending congestion window
+//     |          |    |     |    |     |  | |-------> (ack.quick<<1)|ack.pingpong
+//     |          |    |     |    |     |  |---------> Predicted tick of soft clock
+//     |          |    |     |    |     |              (delayed ACK control data)
+//     |          |    |     |    |     |------------> retransmit timeout
+//     |          |    |     |    |------------------> location of socket in memory
+//     |          |    |     |-----------------------> socket reference count
+//     |          |    |-----------------------------> inode
+//     |          |----------------------------------> unanswered 0-window probes
+//     |---------------------------------------------> uid
+//
+// Source: https://www.kernel.org/doc/Documentation/networking/proc_net_tcp.txt
+
+type scanner struct {
+	file   *os.File
+	reader *bufio.Reader
+
+	hexDecodingBuffer []byte
+}
+
+func newScanner(filepath string) (*scanner, error) {
+	f, err := os.Open(filepath)
+	if err != nil {
+		return nil, err
+	}
+
+	reader := bufio.NewReader(f)
+
+	// Skip header line
+	_, _ = reader.ReadBytes('\n')
+
+	return &scanner{
+		file:              f,
+		reader:            reader,
+		hexDecodingBuffer: make([]byte, 16),
```

Comment:
use a buffer pool?

Reply:
I don't think that would be beneficial in any way 🤔